src/eth/erc4337/types.test.ts

```diff
@@ -76,4 +76,59 @@ describe('types', () => {
       'At path: nonce -- Expected a value of type `EthUint256`, but received: `"0x01"`',
     );
   });
+
+  describe('EthBaseUserOperationStruct', () => {
+    const baseUserOp = {
+      nonce: '0x1',
+      initCode: '0x',
+      callData: '0x70641a22000000000000000000000000',
+      gasLimits: {
+        callGasLimit: '0x58a83',
+        verificationGasLimit: '0xe8c4',
+        preVerificationGas: '0xc57c',
+      },
+      dummyPaymasterAndData: '0x1234',
+      dummySignature: '0x1234',
+    };
+
+    it('is a valid BaseUserOperation', () => {
+      const userOp = {
+        ...baseUserOp,
+        bundlerUrl: 'https://example.com',
+      };
+      expect(() => assert(userOp, EthBaseUserOperationStruct)).not.toThrow();
+    });
+
+    it('has an invalid BaseUserOperation with an incorrect url string', () => {
+      const userOp = {
+        ...baseUserOp,
+        bundlerUrl: 'random string',
+      };
+      expect(() => assert(userOp, EthBaseUserOperationStruct)).toThrow(
+        'At path: bundlerUrl -- Expected a value of type `string`, but received: `"random string"`',
```

Comment:
I was about to say we could re-use `userOp.bundlerUrl` using template-literals here, but that would require to escape backticks, which could be a bit more "cryptic" to read.

I don't have a strong opinion but still wanted to point that out in case you think it's better 😄